spec/Classes.tex

```diff
@@ -663,73 +604,456 @@ \subsection{User-Defined Constructors}
+
+  proc init(x: real, y: real) {
+    this.x = x;
+    this.y = y;
+    // compiler inserts initialization for 'max' and 'msg'
+
+    this.complete(); // 'this' is now considered to be fully initialized
+
+    this.verify();
+    writeln(this);
+  }
+
+  proc init(msg : string) {
+    // compiler inserts initialization for fields 'x', 'y', and 'max'
+    this.msg = msg;
+
+    // Illegal: this.verify();
+    // Implicit 'this.complete();'
+  }
+
+  proc verify() {
+    if x > max || y > max then
+      halt("LabeledPoint out of bounds!");
+  }
+}
+
+var A = new LabeledPoint(1.0, 2.0);
+var B = new LabeledPoint("Origin");
+\end{chapel}
+\begin{chapelpost}
+writeln(B);
+\end{chapelpost}
+\begin{chapelcompopts}
+--no-warnings
+\end{chapelcompopts}
+\begin{chapeloutput}
+{x = 1.0, y = 2.0, max = 10.0, msg = Unlabeled}
+{x = 0.0, y = 0.0, max = 10.0, msg = Origin}
+\end{chapeloutput}
+
+The first initializer leverages the \chpl{complete} method to initialize the
+remaining fields and to allow for the usage of the \chpl{verify} method.
+Calling the \chpl{verify} method or passing \chpl{this} to \chpl{writeln}
+before the \chpl{complete} method is called would result in a compile-time
+error.
+
+The second initializer exists to emphasize the rule that even though all
+fields are initialized after the initialization of the \chpl{msg} field, the
+compiler does not consider the type initialized until the \chpl{complete}
+method is called. If the second initializer tried to invoke the \chpl{verify}
+method, a compile-time error would be issued.
+\end{chapelexample}
+
+\subsubsection{Invoking Other Initializers}
+\label{Invoking_Other_Initializers}
+\index{classes!initializers!user-defined!other-initializers}
+
+In order to allow for code-reuse, an initializer may invoke another initializer
+implemented for the same type. Because the invoked initializer must operate on
+completely uninitialized memory, a compile-time error will be issued for field
+initialization before a call to \chpl{init}. Because each initializer
+either explicitly or implicitly invokes the \chpl{complete} method, all fields
+and methods may be used after such a call to \chpl{init}.
+
+\begin{chapelexample}{thisDotInit.chpl}
+In the following code:
+\begin{chapel}
+class Point3D {
+  var x, y, z : real;
+
+  proc init(x: real, y: real, z: real) {
+    this.x = x;
+    this.y = y;
+    this.z = z;
+    // implicit 'this.complete();'
+  }
+
+  proc init(u: real) {
+    this.init(u, u, u);
+    writeln(this);
+  }
+}
+
+var A = new Point3D(1.0);
+\end{chapel}
+\begin{chapelcompopts}
+--no-warnings
+\end{chapelcompopts}
+\begin{chapeloutput}
+{x = 1.0, y = 1.0, z = 1.0}
+\end{chapeloutput}
+The second initializer leverages the first initializer to initialize all fields
+with the same value. After the \chpl{init} call the type is fully initialized,
+the \chpl{complete} method has been invoked, and so \chpl{this} can be passed
+to the \chpl{writeln} function.
```

Comment:
Assuming the "parent initializer call" section exists (which I suspect it does but I haven't finished reading yet), you might want to call it out and link to it at the end of this section